iree/compiler/Conversion/HLOToLinalg/HLOToLinalgOnTensors.cpp

```diff
@@ -438,13 +534,29 @@ void populateHLOToLinalgOnTensorsConversionPatterns(
                   ConcatenateOpConversion, DepthwiseConvOpConversion>(context);
 }
 
-std::unique_ptr<OperationPass<FuncOp>> createHLOToLinalgOnTensorsPass() {
-  return std::make_unique<ConvertHLOToLinalgOnTensorsPass>();
+std::unique_ptr<OperationPass<FuncOp>> createHLOToLinalgOnTensorsPass(
+    bool useLinalgOnTensorsPath) {
+  return std::make_unique<ConvertHLOToLinalgOnTensorsPass>(
+      useLinalgOnTensorsPath);
+}
+
+std::unique_ptr<OperationPass<FuncOp>>
+createHLOToLinalgOnTensorsPassExperimental() {
+  return std::make_unique<ConvertHLOToLinalgOnTensorsPassExperimental>();
 }
 
 static PassRegistration<ConvertHLOToLinalgOnTensorsPass> legalize_pass(
     "iree-codegen-hlo-to-linalg-on-tensors",
-    "Convert from XLA-HLO ops to Linalg ops on tensors");
+    "Convert from XLA-HLO ops to Linalg ops on tensors",
+    []() { return std::make_unique<ConvertHLOToLinalgOnTensorsPass>(); });
+
+static PassRegistration<ConvertHLOToLinalgOnTensorsPassExperimental>
+    legalize_pass_linalg_on_tensors(
+        "iree-codegen-hlo-to-linalg-on-tensors-experimental",
+        "Convert from XLA-HLO ops to Linalg ops on tensors", []() {
+          return std::make_unique<
+              ConvertHLOToLinalgOnTensorsPassExperimental>();
+        });
```

Comment:
FYI you don't need to create another pass class `ConvertHLOToLinalgOnTensorsPassExperimental`, in this lambda you can directly do `return std::make_unique<ConvertHLOToLinalgOnTensorsPass>(true);`

But also, if `useLinalgOnTensorsPath` was a pass option, it would be available from the command line

Reply:
I am trying different combinations to make exactly that work. Maybe its the way `iree-opt` is setup, I cannot get the flag exposed. I keep getting flag not recognized error. I am guessing its the way pass registration is setup up in IREE.

With respect to instance specific pass options, we moved away from using that. We ended up needing one flag on the pass, a similar flag on the pipeline and a third on the binary. The first two was only needed for lit testing, the last for e2e execution. So we have moved to using a single option that is used to control options everywhere. I think I have raised this issue a few times, but it seems there is no way around that.